geode-lucene/src/integrationTest/java/org/apache/geode/cache/lucene/ToplevelObjectSerializerOnGrandSubclassIntegrationTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.lucene;
+
+import static org.apache.geode.cache.lucene.test.LuceneTestUtilities.INDEX_NAME;
+import static org.apache.geode.cache.lucene.test.LuceneTestUtilities.REGION_NAME;
+
+import org.apache.lucene.analysis.core.KeywordAnalyzer;
+import org.junit.Test;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionShortcut;
+
+public class ToplevelObjectSerializerOnGrandSubclassIntegrationTest
+    extends NestedObjectSerializerOnGrandSubclassIntegrationTest {
+  protected Region createRegionAndIndex() {
+    luceneService.createIndexFactory()
+        .addField("name").addField("phoneNumbers").addField("myHomePages.content")
+        .addField("contacts.name").addField("contacts.email", new KeywordAnalyzer())
+        .addField("contacts.phoneNumbers").addField("contacts.address")
+        .addField("contacts.homepage.content").addField("contacts.homepage.id")
+        .addField(LuceneService.REGION_VALUE_FIELD).create(INDEX_NAME, REGION_NAME);
+
+    Region region = createRegion(REGION_NAME, RegionShortcut.PARTITION);
+    return region;
+  }
+
+  // override tests using nested object or collection
+  @Test
+  public void queryOnThreeLayerField() {}
+
+  @Test
+  public void queryOnTopLevelStringCollectionField_OR_OneExist() {}
+
+  @Test
+  public void queryOnContactAddressWithStandardAnalyzer() {}
+
+  @Test
+  public void queryOnContactEmailWithAnalyzer() {}
+
+  @Test
+  public void queryOnContactNameWithExpression() {}
+
+  @Test
+  public void queryOnTopLevelObjectCollectionField_AND_BothExist() {}
+
+  @Test
+  public void queryOnContactNameWithExactMath() {}
+
+  @Test
+  public void queryOnTopLevelObjectCollectionField_OR_OneExist() {}
+
+  @Test
+  public void queryOnTopLevelStringCollectionField_AND_BothExist() {}
```

Comment:
The fact that is test is overriding so many methods with empty implementations looks like it means maybe this shoudl not extend NestedObjectSerializerOnGrandSubclassIntegrationTest. Maybe you need a different parent class for both tests?